### PR TITLE
Test with guzzle/PSR7 2.0

### PR DESCRIPTION
**Pull request by Nyholm**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -43,6 +43,10 @@ jobs:
       max-parallel: 10
       matrix:
         php: ['7.2', '7.3', '7.4', '8.0']
+        psr7: ['^1.7']
+        include:
+          - php: '7.4'
+            psr7: '2.0.x-dev'
```
Comment (GrahamCampbell) [inline .github/workflows/ci.yml:49]: `^2.0@dev`?

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: composer.json [hunk 2]
```diff
@@ -69,5 +69,7 @@
         "psr-4": {
             "GuzzleHttp\\Tests\\": "tests/"
         }
-    }
+    },
+    "prefer-stable": true,
+    "minimum-stability": "dev"
```
Comment (GrahamCampbell) [inline composer.json:74]: Why is that needed?

Comment (GrahamCampbell) [reply composer.json:74]: Better to just have the dev on the single dependency we want, otherwise now we run the risk of accidentally needing other dev deps and the CI not catching it.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: composer.json (hunk 2 above)
Comment (Nyholm) [reply composer.json:74]: This allows us to do `vendor/package:^3.0` before 3.0 is released. 

That will come in handy when Guzzle officially supports `guzzlehttp/psr7:2.0` which should happen before 2.0 is released. 

It also has the benefit that we dont need to update the CI when 2.0 is released.

Comment (GrahamCampbell) [reply composer.json:74]: Doing `@dev` on the end of the individual version constraint we want also allows us to do that.

**Comment by Nyholm** [comment]
This PR adding two new jobs. One that will test with `@alfa`, `@beta` and stable releases of ^2.0. And another job that will always test with the latest.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -43,6 +43,12 @@ jobs:
       max-parallel: 10
       matrix:
         php: ['7.2', '7.3', '7.4', '8.0']
+        psr7: ['^1.7']
+        include:
+          - php: '7.4'
+            psr7: '^2.0'
+          - php: '7.4'
+            psr7: 'dev-master'
```
Comment (GrahamCampbell) [inline .github/workflows/ci.yml:51]: `^2.0@dev` does this already, and is robust again branch renames.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 3 above)
Comment (Nyholm) [reply .github/workflows/ci.yml]: The PR is updated

**Review by GrahamCampbell — APPROVED** [review]

**Comment by Nyholm** [comment]
Thank you
